hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/H2LegacyDialect.java

```diff
@@ -0,0 +1,707 @@
+
+		// Prior to 1.4.200 there was no support for 'current value for sequence_name'
+		// After 2.0.202 there is no support for 'sequence_name.nextval' and 'sequence_name.currval'
+		ansiSequence = version.isSameOrAfter( 1, 4, 200 );
+
+		// Prior to 1.4.200 the 'cascade' in 'drop table' was implicit
+		cascadeConstraints = version.isSameOrAfter( 1, 4, 200 );
+		// 1.4.200 introduced changes in current_time and current_timestamp
+		useLocalTime = version.isSameOrAfter( 1, 4, 200 );
+
+		if ( version.isSameOrAfter( 1, 4, 32 ) ) {
+			this.sequenceInformationExtractor = version.isSameOrAfter( 1, 4, 201 )
+					? SequenceInformationExtractorLegacyImpl.INSTANCE
+					: SequenceInformationExtractorH2DatabaseImpl.INSTANCE;
+			this.querySequenceString = "select * from INFORMATION_SCHEMA.SEQUENCES";
+		}
+		else {
+			this.sequenceInformationExtractor = SequenceInformationExtractorNoOpImpl.INSTANCE;
+			this.querySequenceString = null;
+		}
+	}
+
+	private static DatabaseVersion parseVersion(DialectResolutionInfo info) {
+		return DatabaseVersion.make( info.getMajor(), info.getMinor(), parseBuildId( info ) );
+	}
+
+	private static int parseBuildId(DialectResolutionInfo info) {
+		final String databaseVersion = info.getDatabaseVersion();
+		if ( databaseVersion == null ) {
+			return 0;
+		}
+
+		final String[] bits = databaseVersion.split("[. ]");
+		return bits.length > 2 ? Integer.parseInt( bits[2] ) : 0;
+	}
+
+	@Override
+	public boolean getDefaultNonContextualLobCreation() {
+		// http://code.google.com/p/h2database/issues/detail?id=235
+		return true;
+	}
+
+	@Override
+	public boolean supportsStandardArrays() {
+		return getVersion().isSameOrAfter( 2 );
+	}
+
+	@Override
+	protected String columnType(int sqlTypeCode) {
+		switch ( sqlTypeCode ) {
+			// prior to version 2.0, H2 reported NUMERIC columns as DECIMAL,
+			// which caused problems for schema update tool
+			case NUMERIC:
+				return getVersion().isBefore( 2 ) ? columnType( DECIMAL ) : super.columnType( sqlTypeCode );
+			case NCHAR:
+				return columnType( CHAR );
+			case NVARCHAR:
+				return columnType( VARCHAR );
+		}
+		return super.columnType( sqlTypeCode );
+	}
+
+	@Override
+	protected String castType(int sqlTypeCode) {
+		switch ( sqlTypeCode ) {
+			case CHAR:
+			case NCHAR:
+				return "char";
+			case VARCHAR:
+			case NVARCHAR:
+			case LONG32VARCHAR:
+			case LONG32NVARCHAR:
+				return "varchar";
+			case BINARY:
+			case VARBINARY:
+			case LONG32VARBINARY:
+				return "varbinary";
+		}
+		return super.castType( sqlTypeCode );
+	}
+
+	@Override
+	protected void registerColumnTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
+		super.registerColumnTypes( typeContributions, serviceRegistry );
+		final DdlTypeRegistry ddlTypeRegistry = typeContributions.getTypeConfiguration().getDdlTypeRegistry();
+
+		if ( getVersion().isBefore( 2 ) ) {
+			ddlTypeRegistry.addDescriptor( new DdlTypeImpl( ARRAY, "array", this ) );
+		}
```

Comment:
The same problem here, I don't understand why legacy dialect needs a check for H2 2.\*.\*.